pkg/cloud/azure/actuators/machine_scope.go

```diff
@@ -392,3 +412,57 @@ func getEnvironment(m *MachineScope) (*azure.Environment, error) {
 
 	return &env, nil
 }
+
+// getTagList is for getting the merged tag list present in
+// AzureMachineProviderSpec and Infrastructure.Status. Both
+// will have the same tags, unless AzureMachineProviderSpec.Tags
+// has been modified to add/update/delete any tag. Merge will be
+// necessary in delete case, to honour the user-defined tags
+// in Infrastructure.Status and those to newly created resources.
+func getTagList(ocpTags, infraStatusTags, machineSpecTags map[string]string) (map[string]*string, error) {
+	if machineSpecTags == nil && infraStatusTags == nil {
+		return *to.StringMapPtr(ocpTags), nil
+	}
+
+	tags := make(map[string]*string)
+	// copy user defined tags in Infrastructure.Status
+	for k, v := range infraStatusTags {
+		tags[k] = to.StringPtr(v)
+	}
+
+	// merge tags present in Infrastructure.Status with
+	// the tags configured in AzureMachineProviderSpec, with
+	// precedence given to those in AzureMachineProviderSpec
+	// for new or updated tags.
+	for k, v := range machineSpecTags {
+		tags[k] = to.StringPtr(v)
+	}
+
+	// Automation, Content Delivery Network, DNS Azure resources can have a
+	// maximum of 15 tags. OpenShift is reserved with 5 tags for internal use,
+	// allowing 10 tags for user configuration. But this operator does not create
+	// any of the above resources and supports configuring additional tags, hence
+	// increasing the user tags max limit to 45.
+	if len(tags) > 45 {
+		return nil, fmt.Errorf("user configured tags cannot be more than 45, " +
+			"infrstructure.status.resourceTags and Machine.Spec.ProviderSpec.Tags put together")
+	}
```

Comment:
What happens if you have an OCP tag already in the machine or infra tags, I think you need to iterate through any keys in `ocpTags` and delete them from this map before you run this check right?